Modix/Modules/WikipediaModule.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Discord;
+using Discord.Commands;
+using Modix.Services.StackExchange;
+using System;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Modix.Modules
+{
+    [Name("Wikipedia"), Summary("Search Wikipedia from Discord!")]
+    public class WikipediaModule : ModuleBase
+    {
+        [Command("wiki"), Summary("Returns a Wikipedia page result matching the search phrase.")]
+        public async Task Run([Remainder] string phrase)
+        {
+            var response = await new WikipediaService().GetWikipediaResultsAsync(phrase);
+
+            if (response != null && response.Query != null && response.Query.Pages.Any())
+            {
+                var messageBuilder = new StringBuilder();
+                foreach (var page in response.Query.Pages.Values)
+                {
+                    if (string.IsNullOrWhiteSpace(page.Extract))
+                    {
+                        await ReplyAsync($"Failed to find anything for `!wiki {phrase}`.");
+                    }
+                    else
+                    {
+                        messageBuilder.AppendLine(page.Extract);
+                    }
+                }
+
+                var message = messageBuilder.ToString();
+
+                if (message.Length > 0)
+                {
+                    if (message.Length > DiscordConfig.MaxMessageSize)
+                    {
+                        // How many batches do we need to send?
+                        // IE: 5000 / 2000 = 2.5
+                        // Round up = 3
+                        decimal batchCount = Math.Ceiling(decimal.Divide(message.Length, DiscordConfig.MaxMessageSize));
+                        int cursor = 0;
```

Comment:
What exactly is a cursor in that case? Im really tired, so i might just be a little to stupid, but i just cant get behind what its doing. Maybe rename it?